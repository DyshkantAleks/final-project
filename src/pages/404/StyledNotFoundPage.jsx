import styled from 'styled-components';
import { device } from '../../styles/breakpoints/breakpoints';

export const ContainerFor404 = styled.div`
background-color: #f7f8f9;
width: 100%;
height: 100vh;
display: flex;
align-items: center;
justify-content: center;
flex-direction: column;
`;

export const Image404 = styled.img`
width: 100%;
height: 100%;
`;

export const ContainerImage404 = styled.div`
width: 100%;
  @media ${device.tabletM}{
    width: 60rem;
    margin: 0 auto;
  };
`;

export const Text404 = styled.p`
font-size: 2rem;
font-weight: 600;
text-align: center;
`;

export const Button404 = styled.div`
font-size: 1.6rem;
background-color: #6F909A;
border: .1rem solid #6F909A;
border-radius: 1rem;
padding: 1rem 1.5rem;
color: #FFFFFF;
margin: 0 auto;
font-weight: 600;
width: fit-content;
`;

export const TextContainer404 = styled.div`
display: flex;
flex-direction: column;
`;