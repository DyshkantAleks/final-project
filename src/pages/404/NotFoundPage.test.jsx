import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';

import { NotFoundPage } from './index';
import { Button404, ContainerFor404, ContainerImage404, Image404, Text404 } from './StyledNotFoundPage';

const setUp = (props = {}) => shallow(<NotFoundPage {...props} />);

describe('NotFoundPage component', () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });
  it('should be render with wrapper', () => {
    const wrapper = component.find(ContainerFor404);
    expect(wrapper.length).toEqual(1);
  });
  it('should be render with container for image', () => {
    const containerImg = component.find(ContainerImage404);
    expect(containerImg.length).toEqual(1);
  });
  it('should be render with image', () => {
    const image = component.find(Image404);
    expect(image.length).toEqual(1);
    expect(image.prop('src')).toEqual('https://res.cloudinary.com/dg-home/image/upload/v1604486670/General/404_f3bscz.gif');
  });
  it('should be render with text', () => {
    const text = component.find(Text404);
    expect(text.length).toEqual(1);
    expect(text.text()).toEqual('Упс.... Кажется мы потеряли эту страницу :(');
  });
  it('should be redirect on homepage', () => {
    const link = component.find(Link);
    expect(link.prop('to')).toEqual('/');
  })
  it('should be render with button', () => {
    const button = component.find(Button404);
    expect(button.text()).toEqual('Вернуться на главную');
  });
  it('snapshot <NotFoundPage />', () => {
    const wrapper = component.find(ContainerFor404);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});