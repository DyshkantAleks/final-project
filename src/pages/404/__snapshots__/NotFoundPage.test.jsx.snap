// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component snapshot <NotFoundPage 1`] = `
"<styled.div>
  <styled.div>
    <styled.div>
      <styled.img src=\\"https://res.cloudinary.com/dg-home/image/upload/v1604486670/General/404_f3bscz.gif\\" />
    </styled.div>
    <styled.div>
      <styled.p>
        Упс.... Кажется мы потеряли эту страницу :(
      </styled.p>
      <Link to=\\"/\\">
        <styled.div>
          Вернуться на главную
        </styled.div>
      </Link>
    </styled.div>
  </styled.div>
</styled.div>"
`;